pom.xml

```diff
@@ -955,9 +996,10 @@ under the License.
 					<reuseForks>${flink.reuseForks}</reuseForks>
 					<systemPropertyVariables>
 						<forkNumber>0${surefire.forkNumber}</forkNumber>
+						<mvn.forkNumber>0${surefire.forkNumber}</mvn.forkNumber>
 						<log4j.configuration>${log4j.configuration}</log4j.configuration>
 					</systemPropertyVariables>
-					<argLine>-Xms256m -Xmx800m -Dmvn.forkNumber=${surefire.forkNumber} -XX:-UseGCOverheadLimit</argLine>
+					<!-- Do not define argLine here. See http://www.eclemma.org/jacoco/trunk/doc/prepare-agent-mojo.html -->
```

Comment:
That's bound to cause problems because a lot of code assumes these arguments here. We shouldn't touch them and instead do as the documentation you linked suggested:



One of the ways to do this in case of maven-surefire-plugin - is to use syntax for late property evaluation:

```xml
<argLine>@{argLine} -Xms256m -Xmx800m -Dmvn.forkNumber=${surefire.forkNumber} -XX:-UseGCOverheadLimit</argLine>
```